### ISSUE TITLE
Match expressions in the REPL match in the wrong order!

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#706]
Try with the input context:
```
   define asset Product {
     id : String
   }
   define asset Car extends Product {
     range : String
   }
  define constant products = [
    Product{ id : "Blender" },
    Car{ id : "Batmobile", range: "Infinite" },
    Product{ id : "Cup" }
  ]
```

One the expression:
```
return
  foreach p in products return
  match p
    with let x : Product then "Product (" ++ x.id ++ ")"
    with let x : Car then "Car (" ++ x.id ++ ") with range " ++ x.range
    else "Not a product"
```
Which results in `["Product (Blender)", "Car (Batmobile) with range Infinite", "Product (Cup)"] : String[]`

While the following expression:
```
return
  foreach p in products return
  match p
    with let x : Car then "Car (" ++ x.id ++ ") with range " ++ x.range
    with let x : Product then "Product (" ++ x.id ++ ")"
    else "Not a product"
```
results in `["Product (Blender)", "Product (Batmobile)", "Product (Cup)"] : String[]`

**Comment by jeromesimeon** [comment]
This bug does not affect the JavaScript backend, which returns the correct result:
```
bash-3.2$ cat examples/matchtest/logic/logic.ergo 
namespace org.accordproject.matchtest

define constant products = [
    Product{ id : "Blender" },
    Car{ id : "Batmobile", range: "Infinite" },
    Product{ id : "Cup" }
]

contract OptTest over TemplateModel {
  // Simple Clause
  clause test(request : Request) : Response {
		let infos =
			foreach p in products return
    		match p
        with let x : Car then "Car (" ++ x.id ++ ") with range " ++ x.range
        with let x : Product then "Product (" ++ x.id ++ ")"
        else "Not a product";
		
    return Response{
		  test1:infos
		}
  }
}
bash-3.2$ ergo trigger --template ./examples/matchtest --data ./examples/matchtest/data.json --request ./examples/matchtest/request.json --state ./examples/matchtest/state.json
14:19:57 - info:
{
  "clause": "orgXaccordprojectXmatchtestXOptTest",
  "request": {
    "$class": "org.accordproject.matchtest.Request"
  },
  "response": {
    "$class": "org.accordproject.matchtest.Response",
    "test1": ["Product (Blender)", "Car (Batmobile) with range Infinite", "Product (Cup)"],
    "transactionId": "fadc8956-a9a6-4742-8e52-346d9fe04ccd",
    "timestamp": "2019-11-07T19:19:57.898Z"
  },
  "state": {
    "$class": "org.accordproject.cicero.contract.AccordContractState",
    "stateId": "1"
  },
  "emit": []
}
bash-3.2$ 
```

**Comment by jeromesimeon** [comment]
With fix in PR #706
<img width="941" alt="Screenshot 2019-11-07 at 2 34 13 PM" src="https://user-images.githubusercontent.com/670099/68421192-c7985880-016b-11ea-9836-00acaa44178a.png">

**Comment by jeromesimeon** [comment]
Fixed in #707
